src/yb/yql/cql/cqlserver/cql_rpc.cc

```diff
@@ -359,5 +359,12 @@ CoarseTimePoint CQLInboundCall::GetClientDeadline() const {
   return deadline_;
 }
 
+bool CQLInboundCall::ShouldDrop() const {
+  int64_t rpc_queue_position = GetRpcQueuePosition();
+  int64_t rpc_queue_limit = CQLRequest::ParseRpcQueueLimit(serialized_request_);
+  return rpc_queue_position >= 0 && rpc_queue_limit != 0 &&
+      rpc_queue_position >= rpc_queue_limit;
+}
+
```

Comment:
Looks like it would be easier and more clear to just pass this limit to CallQueued since it already has a check for max_queued_calls_.